src/dbnode/storage/shard.go

```diff
@@ -2006,7 +2019,7 @@ func (s *dbShard) ColdFlush(
 
 		// After writing the full block successfully, update the cold version
 		// in the flush state.
-		nextVersion := s.RetrievableBlockColdVersion(startTime) + 1
+		nextVersion := coldVersion + 1
```

Comment:
Yeah so there is +1 logic here and in the Merge() function. Would prefer we figure out the next version here and then tell the Merge() function to use that

Reply:
Sure.